### PR TITLE
Added batched NP support to payload parser

### PR DESCRIPTION
**Pull request by RobGeada**
Allows passing entire input datasets in one CURL to models within the biased model CI

Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/trustyai-explainability/trustyai-explainability/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `FAI-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.3.x] FAI-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Review by ruivieira — APPROVED** [review]

File: tests/basictests/trustyai_biased_models.sh [hunk 1]
```diff
@@ -124,33 +124,33 @@ function schedule_and_check_request(){
...
+#        for i in {1..5}
+#        do
+#          echo -ne "\rSleeping for rest of day...$((5 - $i))"
+#          sleep 1
+#        done
+#        echo
+#      fi
+#      let "LOOP_IDX++"
+#    done
+#}
```
Comment (danielezonca) [inline tests/basictests/trustyai_biased_models.sh:155]: Leftover?

File: tests/resources/utils/send_data_batch [hunk 2]
```diff
@@ -55,27 +55,32 @@ fi
...
+#
+#  sleep $PAUSE_TIME
+#
+#  if [[ "$LOOP_IDX" -ge $(( $THRESHOLD - 1 )) ]]; then
+#    echo -e "\nAll datapoints sent"
+#    break
+#  fi
+#
+#  let "LOOP_IDX++"
+#done
```
Comment (danielezonca) [inline tests/resources/utils/send_data_batch:81]: Leftover?

**Review by danielezonca — APPROVED** [review]

File: tests/basictests/trustyai_biased_models.sh (hunk 1 above)
Comment (RobGeada) [reply tests/basictests/trustyai_biased_models.sh:155]: these might be useful in the future, so want to preserve them for now

File: tests/resources/utils/send_data_batch (hunk 2 above)
Comment (RobGeada) [reply tests/resources/utils/send_data_batch:81]: see above; this is useful to send datapoints individually if needed

**Pull request merged**